libs/features/mas/src/hydrate.js

```diff
@@ -82,32 +82,39 @@ export function processSubtitle(fields, merchCard, subtitleConfig) {
 }
 
 export function processBackgroundImage(
-    fields,
-    merchCard,
-    backgroundImageConfig,
-    variant,
+  fields,
+  merchCard,
+  backgroundImageConfig,
+  variant,
 ) {
-    if (fields.backgroundImage) {
-        switch (variant) {
-            case 'ccd-slice':
-                if (backgroundImageConfig) {
-                    merchCard.append(
-                        createTag(
-                            backgroundImageConfig.tag,
-                            { slot: backgroundImageConfig.slot },
-                            `<img loading="lazy" src="${fields.backgroundImage}" />`,
-                        ),
-                    );
-                }
-                break;
-            case 'ccd-suggested':
-                merchCard.setAttribute(
-                    'background-image',
-                    fields.backgroundImage,
-                );
-                break;
-        }
-    }
+  if (fields.backgroundImage) {
+      switch (variant) {
```

Comment:
i sugges to base decision on backgroundImageConfig.
```
if (backgroundImageConfig.tag) {
    // create tag
}
if (backgroundImageConfig.attribute) {
    merchCard.setAttribute(backgroundImageConfig.attribute, fields.backgroundImage);
}
```

then you only need to add this to ccd-suggested.js AEM_FRAGMENT_MAPPING:
```
backgroundImage: { attribute: 'background-image' },
```

this shifts control to the specific variant, leaving hydrate simple and unaware of ccd or adobe home or any other consumer.

Reply:
That's actually a great suggestion. I've made the change and the function is now variant-agnostic. I've also updated UTs accordingly. Thx.